packages/app/src/plugins/webview-preload.js

```diff
@@ -33,24 +14,15 @@ if (micromatch.isMatch(window.location.origin, originsAllowed) || protocolsAllow
     return p;
   };
 
-  const sendSubscribeToProxy = (channel) => {
-    const o = new Observable(obs => {
-      ipcRenderer.on(`bx-api-subscribe-response-${channel}`, (_, result) => {
-        obs.next(result);
-      });
-    });
+  const addListenerToChannel = (channel, listener) => {
+    ipcRenderer.on(`bx-api-subscribe-response-${channel}`, listener);
     setTimeout(() => {
       ipcRenderer.invoke('get-worker-contents-id')
           .then(workerWebContentsId => ipcRenderer.sendTo(workerWebContentsId, 'bx-api-subscribe', channel));
     }, 1);
-    return o;
   };
```

Comment:
Why was Observable pattern removed from here? It seems like it doesn't help much fixing the issue, and it makes using it in tsx fix a little bit more cumbersome.

I also just noticed that it should probably be updated the following to avoid memory leaks:
```ts
 const o = new Observable(obs => {
  const fn = (_, result) => {
    obs.next(result);
  };
  ipcRenderer.on(`bx-api-subscribe-response-${channel}`, fn);
  
  return () => {
    ipcRenderer.off(`bx-api-subscribe-response-${channel}`, fn);
  }
});
```

Reply:
The code inside `contextBridge` block is executing inside WebView and has no access to our libraries. There is no rxjs inside.

Reply:
Okay I see, let me complete my review then